processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -374,12 +374,26 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 	return tags
 }
 
-func (c *WatchClient) extractPodContainersAttributes(pod *api_v1.Pod) map[string]*Container {
-	containers := map[string]*Container{}
-
-	if c.Rules.ContainerImageName || c.Rules.ContainerImageTag {
+func (c *WatchClient) extractPodContainersAttributes(pod *api_v1.Pod) PodContainers {
+	containers := PodContainers{
+		ByID:   map[string]*Container{},
+		ByName: map[string]*Container{},
+	}
+	if !needContainerAttributes(c.Rules) {
+		return containers
+	}
+	ifRule := func(rule bool, v string) string {
+		if rule {
+			return v
+		}
+		return ""
+	}
+	switch {
+	case c.Rules.ContainerImageName,
```

Comment:
nit: I don't think this switch + one case approach is used across the repo. Let's not introduce new idioms